src/main/java/com/uid2/shared/store/reader/RotatingS3KeyProvider.java

```diff
@@ -84,21 +82,22 @@ public Collection<S3Key> getKeysForSite(Integer siteId) {
                 .collect(Collectors.toList());
     }
 
+    public List<S3Key> getKeysForSiteFromMap(int siteId) {
+        return siteToKeysMap.getOrDefault(siteId, new ArrayList<>());
+    }
+
     public S3Key getEncryptionKeyForSite(Integer siteId) {
         Collection<S3Key> keys = getKeysForSite(siteId);
         if (keys.isEmpty()) {
             throw new IllegalStateException("No S3 keys available for encryption for site ID: " + siteId);
-        } else {
-            Map<Integer, S3Key> allKeys = getAll();
-            S3Key largestKey = null;
-            for (S3Key key : allKeys.values()) {
-                if (key.getSiteId() == siteId) {
-                    if (largestKey == null || key.getId() > largestKey.getId()) {
-                        largestKey = key;
-                    }
-                }
-            }
-            return largestKey;
         }
+        return findLargestKey(keys);
     }
+
+    S3Key findLargestKey(Collection<S3Key> keys) {
```

Comment:
Do we use this anywhere besides the above function? Seems like we can just get rid of it

Reply:
`findLargestKey` is currently being used by `getEncryptionKeyForSite`. Right now, I am using a temporary key selection method by simply choosing the largest key for a specific siteId. I planned to use this temporary key selection logic until someone started work on key rotation.